server/controllers/upload.js

```diff
@@ -63,3 +65,67 @@ exports.upload = asyncHandler(async (req, res) => {
     }
   });
 });
+
+const AWSPic = multer({
+  storage: multerS3({
+    s3: s3,
+    bucket: process.env.S3_BUCKET,
+    key: function (req, file, cb) {
+      cb(
+        null,
+        path.basename(file.originalname, path.extname(file.originalname)) +
+        "-" +
+        Date.now() +
+        path.extname(file.originalname)
+      );
+    },
+  }),
+  fileFilter: function (req, file, cb) {
+    checkFileType(file, cb);
+  },
+}).single("image")
+
+exports.uploadProfilePic = asyncHandler(async (req, res, next) => {
+  AWSPic(req, res, async (error) => {
+    if (error) {
+      res.json({
+        error: error
+      });
+    } else {
+      if (req.file === undefined) {
+        console.log("Error: No File Selected");
+      } else {
+        try {
+          const { profilePic } = await User.findByIdAndUpdate(req.user.id, {
+            $set: {
+              profilePic: req.file.location
+            }
+          }, { new: true })
+          res.json(profilePic);
+        } catch (err) {
+          res.status(500).json(err);
+        }
+      }
+    }
+  })
+})
+
+exports.uploadSubmissionPic = asyncHandler(async (req, res, next) => {
```

Comment:
Jeewan has created a controller for uploading multiple images, have you considered using that in this case?

Reply:
I thought about that, but then I wouldn't be able to show the preview of all the images for the submission. I was thinking that if they clicked away from the screen, we could delete those uploaded images from AWS.

Reply:
We talked about this today and decided to leave this as it is for now and delete from AWS if the user navigates away without submitting. Jeewan also knows a way of showing the image preview from their local machine rather than from the AWS link. He said I should push this now and he can create an issue for that when he has time.

Reply:
Thanks for the additional information, we can also discuss this in our meeting tomorrow!

Reply:
Ok. Should I merge this with the main? Is is approved or should I wait til we talk tomorrow?